fs2/prediction_writing_callback.py

```diff
@@ -165,11 +165,15 @@ def on_predict_batch_end(  # pyright: ignore [reportIncompatibleMethodOverride]
             outputs["tgt_lens"],
         ):
             torch.save(
-                data[:unmasked_len].cpu(),
-                self._get_filename(
-                    basename=basename,
-                    speaker=speaker,
-                    language=language,
+                data[:unmasked_len]
+                .cpu()
+                .transpose(0, 1),  # save tensors as [K (bands), T (frames)]
+                str(
```

Comment:
We didn't use to need to cast this Path to a str, I wonder why you do now. In my PR #102, `get_filename` is factored out to the base class, we should have it do `return str(path)` in one place, in `get_filename`, instead of casting everywhere we use it.
Warning: Whichever PR is merged second will have to rebase and resolve conflicts over the use of `get_filename`.

Reply:
rebased and fixed using my suggestion here